src/app/home/components/ExperimentPostContainer/FilterContainer/AreaFilter/hooks/useAreaFilter.ts

```
@@ -14,9 +14,21 @@ const hasSelectedAreaAll = (selectedAreas: Record<string, boolean>) => {
   return AREA_ALL.some((area) => selectedAreas[area]);
 };
 
-const useAreaFilter = () => {
-  const [selectedRegion, setSelectedRegion] = useState<RegionType | null>(null);
-  const [selectedAreas, setSelectedAreas] = useState<Record<string, boolean>>({});
+const getInitialAreas = (initialAreas?: AreaType[]) => {
+  return initialAreas ? initialAreas.reduce((acc, area) => ({ ...acc, [area]: true }), {}) : {};
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)